src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs

```diff
@@ -59,25 +66,19 @@ internal struct ReadStack
         /// </summary>
         public bool UseFastPath;
 
-        private void AddCurrent()
+        /// <summary>
+        /// Ensures that the stack buffer has sufficient capacity to hold an additional frame.
+        /// </summary>
+        private void EnsurePushCapacity()
         {
-            if (_previous == null)
-            {
-                _previous = new List<ReadStackFrame>();
-            }
-
-            if (_count > _previous.Count)
+            if (_stack is null)
             {
-                // Need to allocate a new array element.
-                _previous.Add(Current);
+                _stack = new ReadStackFrame[4];
```

Comment:
4 seems like a reasonable default depth. We any benchmark validations of that performed?

Reply:
This is reusing the default initial capacity for `List<T>`. It seems like a reasonable expected depth for most scenaria so I thought I'd leave it unchanged.